File: qc-core/src/main/java/com/eager/questioncloud/core/domain/payment/infrastructure/QuestionPaymentHistoryMongoRepository.java
package com.eager.questioncloud.core.domain.payment.infrastructure;

import java.util.List;
import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface QuestionPaymentHistoryMongoRepository extends MongoRepository<QuestionPaymentHistoryDocument, Long> {
    List<QuestionPaymentHistoryDocument> findByUserId(Long userId, Pageable pageable);

    int countByUserId(Long userId);
}

File: qc-domain/qc-pay/src/main/kotlin/com/eager/questioncloud/payment/infrastructure/repository/QuestionPaymentHistoryMongoRepository.kt
package com.eager.questioncloud.payment.infrastructure.repository

import com.eager.questioncloud.payment.infrastructure.document.QuestionPaymentHistoryDocument
import org.springframework.data.domain.Pageable
import org.springframework.data.mongodb.repository.MongoRepository

interface QuestionPaymentHistoryMongoRepository :
    MongoRepository<QuestionPaymentHistoryDocument, Long> {
    fun findByUserId(
        userId: Long,
        pageable: Pageable
    ): List<QuestionPaymentHistoryDocument>

    fun countByUserId(userId: Long): Int
}
